### PR TITLE
[ADD] Added website_partner_firstname module to partner-contact modules

### PR DESCRIPTION
**Pull request by PCatinean**
Module for adding First Name / Last Name in website_sale module at checkout

### CODE REVIEW

File: website_partner_contact/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+{
+    'name': 'Partner first name and last name in e-shop',
+    'summary': "Split first name and last name for "
+               "non company partners in e-shop",
```
Comment (guewen) [inline website_partner_contact/__openerp__.py]: Is it really only for non company partners? I see no condition based on if it is a company/contact

Comment (PCatinean) [inline website_partner_contact/__openerp__.py]: I see what you mean, you are right I need to check that

**Comment by pedrobaeza** [comment]
This should be renamed reading the explanation as _website_partner_firstname_

File: website_partner_contact/README.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Partner first name and last name for website_sale
+================================
+
+This module was written to extend the functionality of contacts to support
+having separate last name and first name to the checkout proccess in the frontend
```
Comment (elicoidal) [inline website_partner_contact/README.rst]: s/support/allow 
[...]the separation of first and last name during the check out process in the frontend.

File: website_partner_contact/README.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Credits
+=======
+
+Contributors
+------------
+
+* Paul Catinean <paulcatinean@gmail.com>
+
+Maintainer
+----------
```
Comment (elicoidal) [inline website_partner_contact/README.rst]: add button try me on runbot

File: website_partner_contact/__init__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline website_partner_contact/__init__.py]: Please simplify your headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: website_partner_contact/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    'maintainer': 'Paul Catinean',
+    'category': 'Extra Tools',
+    'depends': [
+        'website_sale',
+        'partner_firstname'
+    ],
+    'data': [
+        'templates.xml',
+    ],
+    'demo': [],
```
Comment (elicoidal) [inline website_partner_contact/__openerp__.py]: Please remove empty keys

File: website_partner_contact/controllers/__init__.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from . import main
```
Comment (elicoidal) [inline website_partner_contact/controllers/__init__.py]: add empty line

**Comment by PCatinean** [comment]
Updates as requested

File: website_partner_firstname/README.rst [hunk 7]
```diff
@@ -0,0 +1,31 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
```
Comment (tafaRU) [inline website_partner_firstname/README.rst:3]: According to https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst you should add here another line of equal signs

File: website_partner_firstname/README.rst [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+Partner first name and last name for website_sale
+================================
```
Comment (tafaRU) [inline website_partner_firstname/README.rst]: Also, equal signs should have same length as module title.

File: website_partner_firstname/templates.xml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+		              <input type="text" name="shipping_firstname" class="form-control" t-att-value="checkout.get('shipping_firstname')"/>
+		          </div>
+		          <div t-attf-class="form-group #{error.get('shipping_lastname') and 'has-error' or ''} col-lg-6">
+		              <label class="control-label" for="shipping_lastname">Last Name</label>
+		              <input type="text" name="shipping_lastname" class="form-control" t-att-value="checkout.get('shipping_lastname')"/>
+		          </div>
+      		</xpath>
+		</template>
+	</data>
+</openerp>
```
Comment (tafaRU) [inline website_partner_firstname/templates.xml]: No newline at the end of file!

File: website_partner_firstname/controllers/main.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    def _get_mandatory_shipping_fields(self):
+        """ Remove name from mandatory fields in order to implement first name
+        and last name fields that are combined in the end"""
+        res = super(
+            WebsitePartnerContact, self)._get_mandatory_shipping_fields()
+        res = [x for x in res if x != 'name']
+        res.extend(['firstname', 'lastname'])
+        return res
+
+
```
Comment (tafaRU) [inline website_partner_firstname/controllers/main.py]: Too many empty lines :wink:

**Comment by rafaelbn** [comment]
Tested in runbot: http://3119631-201-ace8ac.runbot1.odoo-community.org/payment/transfer/feedback
Trying to buy, I arrive until the step payment, after clicking the button "Pay now" it take more than one minute and after that:

> 504 Gateway Time-out

Please check it and let me know of you have the same problem please.

Check travis, is failing.

Thanks

**Comment by rafaelbn** [comment]
Please @PCatinean could you move this PR to https://github.com/OCA/e-commerce, it should be there.
Thanks. I will review it.

**Comment by rafaelbn** [comment]
No answer

**Pull request closed without merging**